### ISSUE TITLE
Warn users when translating a command that uses `--follow` + a non-GET method

### ISSUE THREAD

**Issue by ducaale** [ducaale/xh#277]
As reported in #270, there is a difference between how curl and xh behave when following redirects and using a non-GET method:

```sh
$ xh post https://httpbin.org/redirect-to url==https://httpbin.org/get --follow --body
{
    "args": {},
    "headers": {
        "Accept": "*/*",
        "Accept-Encoding": "gzip, deflate, br",
        "Host": "httpbin.org",
        "User-Agent": "xh/0.16.1",
        "X-Amzn-Trace-Id": "Root=1-632ee6dd-26529e365e6971a258aa60c5"
    },
    "origin": "92.40.178.172",
    "url": "https://httpbin.org/get"
}

# Use xh's --curl to translate the prev command to curl (ignoring --body)
$ xh post https://httpbin.org/redirect-to url==https://httpbin.org/get --follow --curl
curl -L -X POST 'https://httpbin.org/redirect-to?url=https%3A%2F%2Fhttpbin.org%2Fget'

# Now, let's run the output from the translation
$ curl -L -X POST 'https://httpbin.org/redirect-to?url=https%3A%2F%2Fhttpbin.org%2Fget'
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>405 Method Not Allowed</title>
<h1>Method Not Allowed</h1>
<p>The method is not allowed for the requested URL.</p>
```

This is due to the `-X` flag, which instructs curl to retain the original HTTP method regardless of the redirect code (see `Why superfluous -X usage is bad` section from https://daniel.haxx.se/blog/2015/09/11/unnecessary-use-of-curl-x/).

Given that people relying on xh's curl translation feature might be caught off guard, we should output a warning when translating a combination of `--follow` and non-GET HTTP method:

```
$ xh post https://httpbin.org/redirect-to url==https://httpbin.org/get --follow --curl
Warning: Using a combination of -X/--request and -L/--location which may cause unintended side effects 

curl -L -X POST 'https://httpbin.org/redirect-to?url=https%3A%2F%2Fhttpbin.org%2Fget'
```

**Comment by jgoday** [comment]
Hi, just playing with this awesome project !

created a very simple PR (#279) related to this item.

**Comment by ducaale** [comment]
Thanks, @jgoday.

I realised that the linked comment wasn't clear enough, so I've updated the issue's description now. Also, do you mind branching off from the `develop` branch for your next round of changes? Thanks

**Comment by jgoday** [comment]
> Thanks, @jgoday.
> 
> I realised that the linked comment wasn't clear enough, so I've updated the issue's description now. Also, do you mind branching off from the `develop` branch for your next round of changes? Thanks

Thanks @ducaale, removed old PR and create new one #280 (pointing to the develop branch)

**Comment by ducaale** [comment]
Resolved in #280
